scripts/gen_mapfiles.py

```diff
@@ -19,19 +19,21 @@
 from pygments.util import docstring_headline
 
 def main():
-    for key in ['lexers', 'formatters']:
+    for key in ['lexers', 'formatters', 'styles']:
         lines = []
         for file in (pygments_package / key).glob('[!_]*.py'):
             module_name = '.'.join(file.relative_to(pygments_package.parent).with_suffix('').parts)
             print(module_name)
             module = import_module(module_name)
             for obj_name in module.__all__:
                 obj = getattr(module, obj_name)
-                desc = (module_name, obj.name, tuple(obj.aliases), tuple(obj.filenames))
+                desc = (module_name, obj.name, tuple(obj.aliases))
                 if key == 'lexers':
-                    desc += (tuple(obj.mimetypes),)
+                    desc += (tuple(obj.filenames), tuple(obj.mimetypes),)
                 elif key == 'formatters':
-                    desc += (docstring_headline(obj),)
+                    desc += (tuple(obj.filenames), docstring_headline(obj),)
+                elif key == 'styles':
+                    desc += (f'{module_name.split(".")[-1]}::{obj_name}',)
```

Comment:
The key is the `object_name`, the main reason I put it in here is to avoid format operations in the `STYLE_MAP` wrapper. The alternative is to create the style map using a dictionary expression and combine things together. Technically you could also get rid of the `module_name.split(".")` here as the `get_style...` function calls `python.styles. + mod` so it just re-assembles it there. Question is how much do we care? I have no problem either way, but this is the lowest cost, highest efficiency one, and we can deprecate/remove `STYLE_MAP` eventually and have a TODO to remove this extra entry then?

Reply:
Ok, another try. Still generates the same `STYLE_MAP`, but uses `STYLES` (more or less) directly instead of `STYLE_MAP`. Not sure this is much better. I think teaching `gen-mappings` to generate an inverse map is more useful.